File: test/src/com/liferay/mobile/sdk/test/portal/PortalVersionTest.java
/**
 * Copyright (c) 2000-2013 Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.sdk.test.portal;

import com.liferay.mobile.android.http.HttpUtil;
import com.liferay.mobile.android.http.PortalVersion;
import com.liferay.mobile.sdk.test.BaseTest;

import java.io.IOException;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * @author Bruno Farache
 */
public class PortalVersionTest extends BaseTest {

	public PortalVersionTest() throws IOException {
		super();
	}

	@Test
	public void getPortalVersion() throws Exception {
		int version = HttpUtil.getPortalVersion(session);

		assertEquals(PortalVersion.V_6_2, version);
	}

}

File: ios/Test/Portal/PortalVersionTest.m
/**
 * Copyright (c) 2000-2014 Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

#import "BaseTest.h"
#import "LRHttpUtil.h"

/**
 * @author Jose Navarro
 */
@interface PortalVersionTest : BaseTest
@end

@implementation PortalVersionTest : BaseTest

- (void)testGetPortalVersion {
	NSError *error;
	int version = [LRHttpUtil getPortalVersion:self.session error:&error];

	XCTAssertNil(error);
	XCTAssertEqual(6200, version);
}

@end